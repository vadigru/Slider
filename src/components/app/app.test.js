import React from "react";
import renderer from "react-test-renderer";

import App from "./app";

it(`Should render App component`, () => {
  const tree = renderer
    .create(
        <App
          width={`1150px`}
          height={`600px`}
          infinite={true}
          caption={true}
          autoplay={true}
          autoplayDelay={5000}
          indicators={true}
          arrows={true}
          adaptiveSlides={true}
          animatedSwipe={true}
          slidesCount={1}
        />,
        {
          createNodeMock: () => {
            return document.createElement(`div`);
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
