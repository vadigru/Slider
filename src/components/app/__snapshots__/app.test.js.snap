// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render App component 1`] = `
<div
  className="slider-wrapper"
>
  <section
    className="slider-settings"
  >
    <div>
      <label
        htmlFor="wrange"
      >
        <span
          className="label-text"
        >
          Width
        </span>
        <span
          className="label-text-output"
        >
          1024
           px
        </span>
        <input
          id="wrange"
          max="1920"
          min="320"
          name="wrange"
          onChange={[Function]}
          type="range"
          value={1024}
        />
      </label>
    </div>
    <div>
      <label
        htmlFor="hrange"
      >
        <span
          className="label-text"
        >
          Height
        </span>
        <span
          className="label-text-output"
        >
          600
           px
        </span>
        <input
          id="hrange"
          max="1080"
          min="320"
          name="hrange"
          onChange={[Function]}
          type="range"
          value={600}
        />
      </label>
    </div>
    <div>
      <label
        htmlFor="animation-time"
      >
        <span
          className="label-text"
        >
          Animation/Speed
        </span>
        <span
          className="label-text-output"
        >
          350
           ms
        </span>
        <input
          id="animation-time"
          max="3000"
          min="100"
          name="animation-time"
          onChange={[Function]}
          type="range"
          value={350}
        />
      </label>
    </div>
    <div>
      <input
        className="visually-hidden"
        defaultChecked={false}
        id="autoplay"
        name="autoplay"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="label-autoplay"
        htmlFor="autoplay"
      >
        <span
          className="label-text"
        >
          Autoplay/Interval
        </span>
        <span
          className="label-text-output"
        >
          3000
           ms
        </span>
        <input
          disabled={true}
          id="autoplay-delay"
          max="10000"
          min="1000"
          name="autoplay-delay"
          onChange={[Function]}
          type="range"
          value={3000}
        />
      </label>
    </div>
    <div>
      <input
        className="visually-hidden"
        defaultChecked={false}
        id="adaptive"
        name="adaptive"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="label-adaptive"
        htmlFor="adaptive"
      >
        <span
          className="label-text"
        >
          Adaptive/Slides
        </span>
        <span
          className="label-text-output"
        >
          1
        </span>
        <input
          disabled={true}
          id="slides-count"
          max={3}
          min="1"
          name="slides-count"
          onChange={[Function]}
          type="range"
          value={1}
        />
      </label>
    </div>
    <div>
      <input
        className="visually-hidden"
        defaultChecked={false}
        id="infinite"
        name="infinite"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="infinite"
      >
        <span
          className="label-text"
        >
          Infinte
        </span>
      </label>
    </div>
    <div>
      <input
        className="visually-hidden"
        defaultChecked={false}
        id="animated-swipe"
        name="animated-swipe"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="animated-swipe"
      >
        <span
          className="label-text"
        >
          Animated swipe
        </span>
      </label>
    </div>
    <div>
      <input
        className="visually-hidden"
        defaultChecked={false}
        id="indicators"
        name="indicators"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="indicators"
      >
        <span
          className="label-text"
        >
          Indicators
        </span>
      </label>
    </div>
    <div>
      <input
        className="visually-hidden"
        defaultChecked={false}
        id="arrows"
        name="arrows"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="arrows"
      >
        <span
          className="label-text"
        >
          Arrows
        </span>
      </label>
    </div>
    <div>
      <input
        className="visually-hidden"
        defaultChecked={false}
        id="caption"
        name="caption"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="caption"
      >
        <span
          className="label-text"
        >
          Caption
        </span>
      </label>
    </div>
  </section>
  <section
    className="slider"
    onClickCapture={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "height": 600,
        "maxWidth": 1024,
        "outline": "2px dashed skyblue",
        "width": "100%",
      }
    }
  >
    <div
      className="slides"
      onAnimationEnd={[Function]}
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "animationDuration": "350ms",
          "animationName": "",
          "animationTimingFunction": "ease",
          "left": "-100%",
        }
      }
    >
      <div
        className="slide slide--no-caption"
        id=""
        style={
          Object {
            "minWidth": "100%",
            "visibility": "hidden",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "lightyellow",
              "display": "flex",
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <h2>
            LAST SLIDE
          </h2>
        </div>
      </div>
      <div
        className="slide slide--no-caption"
        id="slide-1"
        style={
          Object {
            "backgroundImage": "url()",
            "minWidth": "100%",
          }
        }
      >
        <p>
          This is the most simple use.
        </p>
      </div>
      <div
        className="slide slide--no-caption"
        id="slide-2"
        style={
          Object {
            "backgroundImage": "url(img/blackhole.jpg)",
            "minWidth": "100%",
          }
        }
      >
        <img
          alt="Black Hole"
          className="background"
          src="img/blackhole.jpg"
        />
        <div
          className="caption--forced"
        >
          <a
            className="caption-link"
            href="https://www.space.com/where-do-black-holes-lead.html"
          >
            Where Do Black Holes Lead?
          </a>
        </div>
      </div>
      <div
        className="slide slide--no-caption"
        id="slide-3"
        style={
          Object {
            "backgroundImage": "url()",
            "minWidth": "100%",
          }
        }
      >
        <div
          className="content--bg-light"
        >
          <h3>
            Hello, World!
          </h3>
          <img
            alt="World map"
            className="image"
            src="img/world-map.jpg"
          />
        </div>
      </div>
      <div
        className="slide slide--no-caption"
        id="slide-4"
        style={
          Object {
            "backgroundImage": "url(img/iron-man-home.jpg)",
            "minWidth": "100%",
          }
        }
      >
        <img
          alt="Iron Man background image"
          className="background"
          src="img/iron-man-home.jpg"
        />
        <div
          className="content--bg-dark"
        >
          <img
            alt="Iron Man"
            className="avatar"
            heigth="100"
            src="img/iron-man-ava.jpg"
            width="100"
          />
          <h4>
            IRON MAN
          </h4>
          “You can take away my house, all my tricks and toys. But one thing you can't take away... I am Iron Man.”
        </div>
        <div
          className="caption"
        >
          I am the Iron Man
        </div>
      </div>
      <div
        className="slide slide--no-caption"
        id="slide-5"
        style={
          Object {
            "backgroundImage": "url(img/covid19-dashboard.jpg)",
            "minWidth": "100%",
          }
        }
      >
        <img
          alt="COVID-19 dashboard"
          className="background"
          src="img/covid19-dashboard.jpg"
        />
        <div
          className="caption"
        >
          COVID-19 is an infectious disease.
        </div>
        <div
          className="content--bg-light"
        >
          <p>
            Most people who fall sick with COVID-19 will experience mild to moderate symptoms and recover without special treatment.
          </p>
          <a
            className="content-btn"
            href="https://covid19.who.int/"
          >
            COVID-19 Dashboard
          </a>
        </div>
      </div>
      <div
        className="slide slide--no-caption"
        id="slide-6"
        style={
          Object {
            "backgroundImage": "url(img/elbrus.jpg)",
            "minWidth": "100%",
          }
        }
      >
        <img
          alt="Elbrus"
          className="background"
          src="img/elbrus.jpg"
        />
        <div
          className="content--bg-dark"
        >
          <h2>
            Elbrus
          </h2>
        </div>
      </div>
      <div
        className="slide slide--no-caption"
        id="slide-7"
        style={
          Object {
            "backgroundImage": "url()",
            "minWidth": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "lightyellow",
              "display": "flex",
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <h2>
            LAST SLIDE
          </h2>
        </div>
      </div>
      <div
        className="slide slide--no-caption"
        id=""
        style={
          Object {
            "minWidth": "100%",
            "visibility": "hidden",
          }
        }
      >
        <p>
          This is the most simple use.
        </p>
      </div>
    </div>
    
    
  </section>
</div>
`;
